RecoLocalCalo/HGCalRecProducers/plugins/HGCalUncalibRecHitWorkerWeights.cc

```diff
@@ -4,9 +4,62 @@
 #include "FWCore/Framework/interface/Event.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+template< typename DET >
+void configureIt(const edm::ParameterSet& conf, 
+                 HGCalUncalibRecHitRecWeightsAlgo<HGCDataFrame<DET,HGCSample>>& maker) {
+  constexpr char isSiFE[]           = "isSiFE";
+  constexpr char adcNbits[]         = "adcNbits";
+  constexpr char adcSaturation[]    = "adcSaturation";
+  constexpr char tdcNbits[]         = "tdcNbits";
+  constexpr char tdcSaturation[]    = "tdcSaturation";
+  constexpr char tdcOnset[]         = "tdcOnset";
+  constexpr char toaLSB_ns[]        = "toaLSB_ns";
+  
+  if( conf.exists(isSiFE) ) {
+    maker.set_isSiFESim(conf.getParameter<bool>(isSiFE));
+  } else {
+    maker.set_isSiFESim(false);
+  }
+
+  if( conf.exists(adcNbits) ) {
+    uint32_t nBits     = conf.getParameter<uint32_t>(adcNbits);
+    double saturation  = conf.getParameter<double>(adcSaturation);
+    float adcLSB       = saturation/pow(2.,nBits);    
+    maker.set_ADCLSB(adcLSB);
+  } else {
+    maker.set_ADCLSB(-1.);
+  }
+  
+  if( conf.exists(tdcNbits) ) {
+    uint32_t nBits    = conf.getParameter<uint32_t>(tdcNbits);
+    double saturation = conf.getParameter<double>(tdcSaturation);
+    double onset      = conf.getParameter<double>(tdcOnset); // in fC
+    float tdcLSB      = saturation/pow(2.,nBits); 
```

Comment:
actually, nevermind. After you said "bin width" the sense of reason caught up with me on this subject.